doxy

```diff
@@ -1,18 +1,60 @@
 #!/usr/bin/bash
 
-here=`pwd`
-# echo ${here}
-
-doxyfiles=("DoxyLangman" "DoxyRefman"            "DoxyUserman"      "DoxyStlman")
-locations=("langman"     "refman"                "userman"          "stlman")
-books=(    "Bill_Primer" "reference_manual"      "user_manual"      "standard_library_reference")
-names=(    "Bill_Primer" "Bill_Reference_Manual" "Bill_User_Manual" "Bill_Standard_Library_Reference")
+# function to generate PDFs
+genPdf () {
+    # parameters:
+    config=$1
+    book=$2
+    location=$3
+    name=$4
+    
+    # test build folder presence
+    if [ ! -d /tmp/bill ]
+    then
+        mkdir /tmp/bill
+    fi
+    
+    # build if DoxyFile exists
+    doxyfile="dox/books/${book}/${config}"
+    if [ -f ${doxyfile} ]
+    then
+        doxygen ${doxyfile}
+        retVal=$?
+        if [ $retVal -ne 0 ]
+        then
+            echo "Error: exit code ${retVal} on doxygen ${doxyfile}."
+#                 cd $here
+            exit $retVal
+        fi
+        
+        # make pdf
+        pushd /tmp/bill/${location}/latex
+        make
+        retVal=$?
+        if [ $retVal -ne 0 ]
+        then
+            echo "Error: exit code ${retVal} on Make of ${doxyfile}."
+            popd
+            exit $retVal
+        fi
+        popd
+        
+        # move pdf to docs
+        mv /tmp/bill/${location}/latex/refman.pdf docs/${name}.pdf
+    else
+        echo -e "\n========\nInfo: ${doxyfile} does not exist.\n========\n"
+        # future activation
+#         echo -e "\n========\nError: ${doxyfile} does not exist.\n========\n"
+#         popd
+#         exit 1
+    fi
+}
 
-# html forward and config
+# copy html forward and config
```

Comment:
Do these get overwritten by doxygen?  Or can you just git mv them and remove the copies?

Reply:
I do it this way to keep all configuration in dox.

Reply:
I don't like that answer on principle, bu given that we are committing the docs directory for the website....  I guess that makes clear that everything in docs/ is temporary for the website and all source (not so much configuration) is in dox.